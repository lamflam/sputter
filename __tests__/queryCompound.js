const snapshot = require("../testUtil/snapshot");
const fixtures = require("../testUtil/fixtures");

describe("Compound query", () => {
  describe("two attributes", () => {
    snapshot(fixtures.conditional, '[left.name="x"][right.value=1]');
  });

  describe("type and pseudo", () => {
    snapshot(fixtures.conditional, '[left.name="x"]:matches(*)');
  });
});
