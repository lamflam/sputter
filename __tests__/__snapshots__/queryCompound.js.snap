// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compound query two attributes [left.name="x"][right.value=1] 1`] = `
> 1 | if (x === 1) {
    |     ^^^^^^^ BinaryExpression
  2 |   foo();
  3 | } else {
  4 |   x = 2;
`;

exports[`Compound query type and pseudo [left.name="x"]:matches(*) 1`] = `
> 1 | if (x === 1) {
    |     ^^^^^^^ BinaryExpression
  2 |   foo();
  3 | } else {
  4 |   x = 2;
  2 |   foo();
  3 | } else {
> 4 |   x = 2;
    |   ^^^^^ AssignmentExpression
  5 | }
  6 | if ((x == "test" && true) || x) {
  7 |   y = -1;
  4 |   x = 2;
  5 | }
> 6 | if ((x == "test" && true) || x) {
    |      ^^^^^^^^^^^ BinaryExpression
  7 |   y = -1;
  8 | } else if (false) {
  9 |   y = 1;
`;
